### ISSUE TITLE
Terminal's working directory (i.e. for duplicate tab) does not default to the startup directory being used

### ISSUE THREAD

**Issue by TBBle** [microsoft/terminal#8969]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19042.746]
Windows Terminal version (if applicable): 1.6.10272.0
```

# Steps to reproduce

* Get OSC 9;9 working, so that duplicate tab and splitPane+duplicate correctly carry the current working directory (#3158)
* Duplicate a tab in some directory (ctrl+shift+d)
* Before that tab issues OSC 9;9, split-duplicate the pane (alt+shift+d, see below)

The following non-default keybind was used to replicate this:
```jsonc
    {
      "command": {
        "action": "splitPane",
        "split": "auto",
        "splitMode": "duplicate"
      },
      "keys": "alt+shift+d"
    },
```

# Expected behavior

Both panes in the new tab are in the desired working directory

# Actual behavior

The first pane is in the desired working directory, but the second pane is in the default working directory.

# Notes

If I wait between the ctrl+shift+d and alt+shift+d for my profile to load and my prompt to issue the OSC 9;9, then the expected behaviour occurs.

I assume this should be relatively easy to fix, assuming there's some point in the construction process that has access to both `settings.StartingDirectory` and the `ITerminalApi` for the new pane or tab.

**Comment by DHowett** [comment]
Interesting. @skyline75489 I bet if we pre-seed the "last reported working directory" with the starting directory for the pane we will get identical behavior for NON-OSC 9;9 clients and *better* behavior for OSC-9;9 clients.

In short: duplicate-splitting a pane twice starts the third pane in the wrong folder because it tries to copy the unreported starting directory from the second pane which hasn't send its own OSC 9;9 yet.

**Comment by skyline75489** [comment]
I’m not a pane user so I don’t have the whole picture here. But generally I feel it is a fair request to pre-seed the "last reported working directory" with the starting directory for the pane. I’ll dig later.

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Dustin L. Howett <notifications@github.com>
发送时间: Tuesday, February 2, 2021 6:50:27 AM
收件人: microsoft/terminal <terminal@noreply.github.com>
抄送: Chester Liu <skyline75489@outlook.com>; Mention <mention@noreply.github.com>
主题: Re: [microsoft/terminal] Terminal's working directory (i.e. for duplicate tab) does not default to the startup directory being used (#8969)


Interesting. @skyline75489<https://github.com/skyline75489> I bet if we pre-seed the "last reported working directory" with the starting directory for the pane we will get identical behavior for NON-OSC 9;9 clients and better behavior for OSC-9;9 clients.

In short: duplicate-splitting a pane twice starts the third pane in the wrong folder because it tries to copy the unreported starting directory from the second pane which hasn't send its own OSC 9;9 yet.

―

**Comment by TBBle** [comment]
I'm not a pane user either (I was playing with panes for the first time to try out OSC 9;9 ^_^), but I expect dupicate-tab twice would have the same issue.

The suggested pre-seeding fix sounds like what I would expect to work. If it can be done somewhere that isn't specific to the duplicate tab and duplicate pane code-paths, then it would be consistent for any other situations where a terminal is started with an overridden starting directory, e.g. a future "new tab in this directory" command.

One visible change for non-OSC9;9 clients might be that if you started a profile in a different working directory than its default than it's default (I don't know how to do this, does `wt` have a starting-directory override arg?) then duplicate-tab and duplicate-pane would inherit that starting directory. That sounds like it might be an improvement, but might be initially surprising to users.

**Comment by DHowett** [comment]
I don't actually have an "area path" for this. @zadjii-msft, do you have any ideas for the area that means "Terminal, the application, but not the control, but not the UI exactly -- the thing that ties them all together, and the place where OSC 9;9 is truly considered"?

**Comment by DHowett** [comment]
Triaged into 2.0 with help wanted.

**Comment by zadjii-msft** [comment]
We've pretty broadly been using `Area-User Interface` for anything at the `TerminalApp` or `WindowsTerminal` layers, since there's not something more specific. I agree they don't really make sense for this kind of issue. I'll just stick it in UI for now.

**Comment by TBBle** [comment]
I got this wrong last time... The merged fix means we should see this in [1.8](https://github.com/microsoft/terminal/milestone/33), right? (Unless something is wrong and/or the fix is reverted)

**Comment by DHowett** [comment]
I’d likely service this to 1.6 and 1.7, since it’s a low risk quality of life improvement :)
